src/buffer/out/textBuffer.cpp

```diff
@@ -1240,3 +1240,178 @@ std::string TextBuffer::GenHTML(const TextAndColor& rows, const int fontHeightPo
         return {};
     }
 }
+
+// Routine Description:
+// - Generates an RTF document based on the passed in text and color data
+// Arguments:
+// - rows - the text and color data we will format & encapsulate
+// - backgroundColor - default background color for characters, also used in padding
+// - fontHeightPoints - the unscaled font height
+// - fontFaceName - the name of the font used
+// - htmlTitle - value used in title tag of html header. Used to name the application
+// Return Value:
+// - string containing the generated RTF
+std::string TextBuffer::GenRTF(const TextAndColor& rows, const int fontHeightPoints, const std::wstring_view fontFaceName, const COLORREF backgroundColor)
+{
+    try
+    {
+        std::ostringstream rtfBuilder;
+
+        // start rtf
+        rtfBuilder << "{";
+
+        // rtf header
+        rtfBuilder << "\\rtf1\\ansi\\ansicpg1252\\deff0\\nouicompat";

```

Comment:
Just as a personal preference, could we add a comment here for what each of these control words mean? I've been using [this](http://www.biblioscape.com/rtf15_spec.htm) to get a better understanding but I'm confused as to what `\deff0\nouicompat` means.

Reply:
I've been using that too. `\deffN` sets the font at index `N` as the default for the document.
I don't know about `\nouicompat`. This is the header WordPad generates when you create a blank file.

I will add comments in the code.